### DISCUSSION TITLE
Getting windows username without user input

### DISCUSSION THREAD

**Discussion by huntfx** [fastapi/fastapi#7685] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Is it possible to grab the current windows username without requiring the user to do anything? It's in a work environment so I can trust the logged in user is who they say they are, however I don't want to pass it in from the users side as anyone could go and edit it.

I've seen the [Get Current User](https://fastapi.tiangolo.com/tutorial/security/get-current-user/) tutorial, but that's more for an actual login system.

### Additional context
This is an example I found of how to do it in Flask:
```python
def getuser():
    # https://stackoverflow.com/a/48066105/2403000
    user = request.environ.get('REMOTE_USER')
    if user is not None:
        return user

    # https://stackoverflow.com/a/39648839/2403000
    user = request.headers.get('X-Remote-User')
    if user is not None:
        return user
```

**Comment by connebs** [comment, marked as answer]
If there is an `X-Remote-User` header, you can pull it out just like any other header, [as per the docs here about using request headers.](https://fastapi.tiangolo.com/tutorial/header-params/)

As for `flask.Request.environ.REMOTE_USER`, that is just the underlying WSGI environ. I *think* that `environ.REMOTE_USER` is just to specify the user that is authenticated with HTTP basic auth. If you want HTTP basic auth in **FastAPI**, you can follow the [tutorial here.](https://fastapi.tiangolo.com/advanced/security/http-basic-auth/)

**Comment by huntfx** [comment]
Thanks for the info. I guess at this stage then with roughly 200 users, there's no real advantage to using either (other than the IT guy having access to more documentation if setting up flask). I'll just play it by ear with the auth and see what can be done.

**Comment by tiangolo** [comment]
Thanks for your help here @acnebs ! :bowing_man: :cake: 

Thanks for reporting back and closing the issue @Peter92 :+1:
